src/DssGroupedEmergencySpell.sol

```diff
@@ -0,0 +1,129 @@
+        ilkList = _ilks;
+    }
+
+    /// @notice Returns the list of ilks to which the spell is applicable.
+    function ilks() external view returns (bytes32[] memory) {
+        return ilkList;
+    }
+
+    /// @notice Returns the spell description.
+    function description() external view returns (string memory) {
+        // Join the list of ilks into a comma-separated string
+        string memory buf = _bytes32ToString(ilkList[0]);
+        // Start from one because the first item was already added.
+        for (uint256 i = 1; i < ilkList.length; i++) {
+            buf = string.concat(buf, ", ", _bytes32ToString(ilkList[i]));
+        }
+
+        return string.concat(_descriptionPrefix(), " ", buf);
+    }
+
+    /// @notice Converts a bytes32 value into a string.
+    function _bytes32ToString(bytes32 src) internal pure returns (string memory res) {
+        uint256 len = 0;
+        while (src[len] != 0 && len < 32) {
+            len++;
+        }
+        assembly {
+            res := mload(0x40)
+            // new "memory end" including padding (the string isn't larger than 32 bytes)
+            mstore(0x40, add(res, 0x40))
+            // store len in memory
+            mstore(res, len)
+            // write actual data
+            mstore(add(res, 0x20), src)
+        }
+    }
+
+    /// @dev Returns the description prefix to compose the final description.
+    function _descriptionPrefix() internal view virtual returns (string memory);
+
+    /// @inheritdoc DssEmergencySpell
+    function _emergencyActions() internal override {
+        for (uint256 i = 0; i < ilkList.length; i++) {
+            _emergencyActions(ilkList[i]);
+        }
+    }
+
+    /// @notice Executes the emergency actions for all ilks in the batch.
+    /// @dev This is an escape hatch to prevent the spell from being blocked in case it would hit the block gas limit.
+    ///      In case `end` is greater than the ilk registry length, the iteration will be automatically capped.
+    /// @param start The index to start the iteration (inclusive).
+    /// @param end The index to stop the iteration (inclusive).
+    function emergencyActionsInBatch(uint256 start, uint256 end) external {
+        end = end > ilkList.length - 1 ? ilkList.length - 1 : end;
+        require(start <= end, "DssGroupedEmergencySpell/bad-iteration");
+
+        for (uint256 i = start; i <= end; i++) {
+            _emergencyActions(ilkList[i]);
+        }
+    }
+
+    /// @notice Executes the emergency actions for the specified ilk.
+    /// @param _ilk The ilk to set the related Clip breaker.
+    function _emergencyActions(bytes32 _ilk) internal virtual;
+
+    /// @notice Returns whether the spell is done for all ilks or not.
+    /// @return res Whether the spells is done or not.
+    function done() external view returns (bool res) {
+        res = true;
+        for (uint256 i = 0; i < ilkList.length; i++) {
+            res = res && _done(ilkList[i]);
+        }
+    }
```

Comment:
```suggestion
function done() external view returns (bool res) {
    for (uint256 i = 0; i < ilkList.length; i++) {
        if (!_done(ilkList[i])) {
            return false;
        }
    }
    return true;
}
```
Gas optimization: Execution can be halted in the first occurrence of an ilk that is not `done`.